### PR TITLE
Specify era/eraYear.

### LINKED ISSUE #1308: Add extra calendar field getters to the 402 portion of the spec text
I overlooked this before, the `era` getters and the Calendar.era() method have to be present in the 402 portion of the spec. According to the consensus at the 2021-01-19 meeting, so should `eraYear`. All of these return `undefined` for ISO calendar dates.

### PR DESCRIPTION
**Pull request by Ms2ger**
Fixes #1308.

### CODE REVIEW

File: spec/intl.html [hunk 1]
```diff
@@ -927,6 +927,38 @@ <h1>Temporal.Calendar.prototype.dateUntil ( _one_, _two_, _options_ )</h1>
...
+          <p>
+            The `era` method takes one argument _dateOrDateTime_.
+            The following steps are taken:
+          </p>
+          <emu-alg>
+            1. Let _calendar_ be the *this* value.
+            1. Perform ? RequireInternalSlot(_calendar_, [[InitializedTemporalCalendar]]).
+            1. If _calendar_.[[Identifier]] is *"iso8601"*, then
+              1. Return *undefined*.
+            1. Let _era_ be the result of implementation-defined processing of _dateOrDateTime_ and the value of _calendar_.[[Identifier]].
```
Comment (ljharb) [inline spec/intl.html:941]: this is very hand-wavey. can this be elaborated in, or made less imprecise?

File: spec/intl.html [hunk 2]
```diff
@@ -927,6 +927,38 @@ <h1>Temporal.Calendar.prototype.dateUntil ( _one_, _two_, _options_ )</h1>
...
+          <p>
+            The `eraYear` method takes one argument _dateOrDateTime_.
+            The following steps are taken:
+          </p>
+          <emu-alg>
+            1. Let _calendar_ be the *this* value.
+            1. Perform ? RequireInternalSlot(_calendar_, [[InitializedTemporalCalendar]]).
+            1. If _calendar_.[[Identifier]] is *"iso8601"*, then
+              1. Return *undefined*.
+            1. Let _era_ be the result of implementation-defined processing of _dateOrDateTime_ and the value of _calendar_.[[Identifier]].
```
Comment (ljharb) [inline spec/intl.html:957]: also here

File: spec/intl.html [hunk 3]
```diff
@@ -927,6 +927,38 @@ <h1>Temporal.Calendar.prototype.dateUntil ( _one_, _two_, _options_ )</h1>
...
+            1. Return 𝔽(_era_).
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sup-temporal.calendar.prototype.erayear">
+          <h1>Temporal.Calendar.prototype.eraYear ( _dateOrDateTime_ )</h1>
+          <p>
+            The `eraYear` method takes one argument _dateOrDateTime_.
```
Comment (ljharb) [inline spec/intl.html:949]:
```suggestion
            The `eraYear` method takes one argument, _dateOrDateTime_.
```
What kind of argument?

File: spec/intl.html [hunk 4]
```diff
@@ -927,6 +927,38 @@ <h1>Temporal.Calendar.prototype.dateUntil ( _one_, _two_, _options_ )</h1>
           </emu-alg>
         </emu-clause>
 
+        <emu-clause id="sup-temporal.calendar.prototype.era">
+          <h1>Temporal.Calendar.prototype.era ( _dateOrDateTime_ )</h1>
+          <p>
+            The `era` method takes one argument _dateOrDateTime_.
```
Comment (ljharb) [inline spec/intl.html:933]:
```suggestion
            The `era` method takes one argument, _dateOrDateTime_.
```
What kind of argument?

File: spec/intl.html [hunk 5]
```diff
@@ -1187,6 +1219,36 @@ <h1>Temporal.PlainDate.prototype.toLocaleString ( [ _locales_ [ , _options_ ] ]
...
+          <p>
+            `Temporal.PlainDate.prototype.era` is an accessor property whose set accessor function is *undefined*.
+            Its get accessor function performs the following steps:
+          </p>
+          <emu-alg>
+            1. Let _plainDate_ be the *this* value.
+            1. Perform ? RequireInternalSlot(_plainDate_, [[InitializedTemporalDate]]).
+            1. Let _calendar_ be _plainDate_.[[Calendar]].
+            1. Let _method_ be ? Get(_calendar_, *"era"*).
+            1. Return ? Call(_method_, _calendar_, « _plainDate_ »).
```
Comment (ljharb) [inline spec/intl.html:1234]: Any reason not to use Invoke here? (same with all the Get + Calls)

Comment (ptomato) [reply spec/intl.html:1234]: I hadn't encountered Invoke before! ... Good suggestion, I'll open a new issue for that.

File: spec/intl.html [hunk 6]
```diff
@@ -927,6 +927,38 @@ <h1>Temporal.Calendar.prototype.dateUntil ( _one_, _two_, _options_ )</h1>
           </emu-alg>
         </emu-clause>
 
+        <emu-clause id="sup-temporal.calendar.prototype.era">
```
Comment (ptomato) [inline spec/intl.html:930]: These should be `sec-` (here and below) because they don't supersede anything in the 262 part of the document.

**Review by ptomato — APPROVED** [review]
Not sure what to do about the vagueness of the "implementation-defined processing" bits. We can't really specify out all the ICU calendars, but on the other hand I do take the point that it is weird to have methods that return either `undefined` or an implementation-defined result with no explanation.

Would that situation be sufficiently improved when we add the corresponding explanation in the documentation, or does there need to be some explanation in the spec text as well?

**Comment by ljharb** [comment]
I think it needs to be clear in the spec steps.

**Comment by Ms2ger** [comment]
Editorial improvements are still welcome, of course, but I've merged this in the interest of stabilizing the normative parts of the spec.

**Comment by ljharb** [comment]
Given there’s still a number of open review comments, I’m not sure why merging prior to addressing them is stabilizing.

**Pull request merged**